File: app/src/main/java/com/akshayaap/mouseremote/debug/DebugDatabase.java
package com.akshayaap.mouseremote.debug;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class DebugDatabase {
    private final Map<String, ArrayList<LoggMessage>> database;

    public DebugDatabase() {
        database = new HashMap<>();
    }

    public void addMessage(LoggMessage message) {
        ArrayList<LoggMessage> messages = database.get(message.getTag());
        if (messages == null) {
            messages = new ArrayList<>();
            database.put(message.getTag(), messages);
        }
        messages.add(message);
    }

    public ArrayList<LoggMessage> getMessages(String tag) {
        return database.get(tag);
    }

    public ArrayList<LoggMessage> getAllMessages() {
        ArrayList<LoggMessage> list = new ArrayList<>();
        for (String key : database.keySet()) {
            list.addAll(database.get(key));
        }
        return list;
    }
}

File: app/src/main/java/com/akshayaap/touchdroid/debug/DebugDatabase.kt
package com.akshayaap.touchdroid.debug

import com.akshayaap.touchdroid.util.logger.LoggMessage

class DebugDatabase {
    private val database: MutableMap<String, ArrayList<LoggMessage>> = HashMap()

    fun addMessage(message: LoggMessage) {
        val tag = message.tag ?: "default"
        val messages = database.getOrPut(tag) { ArrayList() }
        messages.add(message)
    }

    fun getMessages(tag: String): ArrayList<LoggMessage>? {
        return database[tag]
    }

    fun getAllMessages(): ArrayList<LoggMessage> {
        val list = ArrayList<LoggMessage>()
        for (messages in database.values) {
            list.addAll(messages)
        }
        return list
    }
}
